redis.go

```diff
@@ -288,11 +288,9 @@ func (c *baseClient) withConn(
 		return err
 	}
 
-	defer func() {
-		c.releaseConn(ctx, cn, err)
-	}()
-
-	return fn(ctx, cn)
+	err = fn(ctx, cn)
```

Comment:
I am not sure if this is entirely correct. If `fn` panics the connection will be lost, thus lowering the capacity of the connection pool permanently.

I think a name returned should be used here, and in case of a panic, the connection should always be closed.